azure-mgmt-website/src/main/java/com/microsoft/azure/management/website/HostNameSslBinding.java

```diff
@@ -35,78 +36,230 @@
+            /**
+             * Upload a PFX certificate.
+             * @param pfxFile the PFX certificate file to upload
+             * @param password the password to the certificate
+             * @return the next stage of the hostname SSL binding definition
+             */
             WithSslType<ParentT> withPfxCertificateToUpload(File pfxFile, String password);
-//            WithSslType<ParentT> withNewAppServiceCertificateOrder(CertificateProductType productType, int validYears);
-//            WithSslType<ParentT> withExistingAppServiceCertificate(AppServiceCertificate appServiceCertificate);
+
+            /**
+             * Place a new App Service certificate order to use for the hostname
+             * @param certificateOrderName the name of the certificate order
+             * @param productType the sku of the certificate order
+             * @return the next stage of the hostname SSL binding definition
+             */
+            WithKeyVault<ParentT> withNewAppServiceCertificateOrder(String certificateOrderName, CertificateProductType productType);
+
+            /**
+             * Use a ready-to-use certificate order. This is usually useful for reusing wildcard certificates.
+             * @param certificateOrder the ready-to-use certificate order
+             * @return the next stage of the hostname SSL binding definition
+             */
+            WithSslType<ParentT> withReadyToUseAppServiceCertificateOrder(AppServiceCertificateOrder certificateOrder);
         }
 
+        /**
+         * The stage of hostname SSL binding definition allowing key vault for certificate store to be specified.
+         * @param <ParentT> the return type of the final {@link WithAttach#attach()}
+         */
+        interface WithKeyVault<ParentT> {
+            /**
+             * Store the certificate in an existing vault.
+             * @param vault the existing vault to use
+             * @return the next stage of the hostname SSL binding definition
+             */
+            WithSslType<ParentT> withExistingKeyVault(Vault vault);
+
+            /**
+             * Create a new key vault and store the certificate in it.
+             * @param vaultName the name of the key vault to create
+             * @return the next stage of the hostname SSL binding definition
+             */
+            WithSslType<ParentT> withNewKeyVault(String vaultName);
+        }
+
+        /**
+         * The stage of hostname SSL binding definition allowing SSL type to be specified.
+         * @param <ParentT> the return type of the final {@link WithAttach#attach()}
+         */
         interface WithSslType<ParentT> {
-            WithAttach<ParentT> withSniSSL();
-            WithAttach<ParentT> withIpBasedSSL();
+            /**
+             * Use Server Name Indication (SNI) based SSL.
+             * @return the next stage of the hostname SSL binding definition
+             */
+            WithAttach<ParentT> withSniSsl();
```

Comment:
sincd the other one is called withIpBasedSsl() how about --> withSniBasedSsl() for this to be consistent and a bit clearer?